jaxb-ri/bundles/ri/src/main/resources/bin/schemagen.bat

```diff
@@ -43,45 +43,78 @@ REM
 rem
 rem Make sure that JAXB_HOME and JAVA_HOME are set
 rem
-if not "%JAXB_HOME%" == "" goto CHECKJAVAHOME
+if not "%JAXB_HOME%" == "" goto SETCLASSPATH
 
 rem Try to locate JAXB_HOME
 set JAXB_HOME=%~dp0
 set JAXB_HOME=%JAXB_HOME%\..
-if exist %JAXB_HOME%\lib\jaxb-xjc.jar goto CHECKJAVAHOME
+if exist %JAXB_HOME%\lib\jaxb-xjc.jar goto SETCLASSPATH
 
 rem Unable to find it
 echo JAXB_HOME must be set before running this script
 goto END
 
+:SETCLASSPATH
+set JAXB_PATH=%JAXB_HOME%/mod/jaxb-api.jar;%JAXB_HOME%/mod/jaxb-jxc.jar;%JAXB_HOME%/mod/jaxb-xjc.jar;%JAXB_HOME%/mod/jaxb-runtime.jar;%JAXB_HOME%/mod/stax-ex.jar;%JAXB_HOME%/mod/istack-commons-runtime.jar;%JAXB_HOME%/mod/istack-commons-tools.jar;%JAXB_HOME%/mod/FastInfoset.jar;%JAXB_HOME%/mod/dtd-parser.jar;%JAXB_HOME%/mod/rngom.jar;%JAXB_HOME%/mod/codemodel.jar;%JAXB_HOME%/mod/xsom.jar;%JAXB_HOME%/mod/txw2.jar;%JAXB_HOME%/lib/relaxngDatatype.jar;%JAXB_HOME%/mod/javax.activation.jar
+
+if "%CLASSPATH%" == "" goto NOUSERCLASSPATH
+set LOCALCLASSPATH=%JAXB_PATH%;%CLASSPATH%
+goto CHECKJAVAHOME
+
+:NOUSERCLASSPATH
+set LOCALCLASSPATH=%JAXB_PATH%
+goto CHECKJAVAHOME
+
 :CHECKJAVAHOME
 if not "%JAVA_HOME%" == "" goto USE_JAVA_HOME
 
 set JAVA=java
-goto SETCLASSPATH
+for /f "" %%i in ('where java') do set BINDIR=%%~dpi
+set TOOLS_PATH=%BINDIR%/../lib/tools.jar;%LOCALCLASSPATH%
+goto LAUNCHSCHEMAGEN
 
 :USE_JAVA_HOME
 set JAVA="%JAVA_HOME%\bin\java"
-goto SETCLASSPATH
-
-:SETCLASSPATH
-if "%CLASSPATH%" == "" goto NOUSERCLASSPATH
-set LOCALCLASSPATH=%JAXB_HOME%\lib\jaxb-core.jar;%JAXB_HOME%\lib\jaxb-xjc.jar;%JAXB_HOME%\lib\jaxb-jxc.jar;%JAXB_HOME%\lib\jaxb-impl.jar;%CLASSPATH%
-goto LAUNCHSCHEMAGEN
-
-:NOUSERCLASSPATH
-set LOCALCLASSPATH=%JAXB_HOME%\lib\jaxb-core.jar;%JAXB_HOME%\lib\jaxb-xjc.jar;%JAXB_HOME%\lib\jaxb-jxc.jar;%JAXB_HOME%\lib\jaxb-impl.jar
+set TOOLS_PATH=%JAVA_HOME%/lib/tools.jar;%LOCALCLASSPATH%
 goto LAUNCHSCHEMAGEN
 
 :LAUNCHSCHEMAGEN
+rem Set Java Version
+for /f "tokens=3" %%i in ('java -version 2^>^&1 ^| %SystemRoot%\system32\find.exe "java version"') do (
```

Comment:
Does it look for string "java version" ? I believe it can be also "openjdk version".